core/trino-main/src/main/java/io/trino/operator/table/ExcludeColumns.java

```diff
@@ -0,0 +1,164 @@
+                    GENERIC_TABLE);
+        }
+
+        @Override
+        public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+        {
+            DescriptorArgument excludedColumns = (DescriptorArgument) arguments.get(DESCRIPTOR_ARGUMENT_NAME);
+            if (excludedColumns.equals(NULL_DESCRIPTOR)) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Excluded columns descriptor is null");
+            }
+            Descriptor excludedColumnsDescriptor = excludedColumns.getDescriptor().orElseThrow();
+            if (excludedColumnsDescriptor.getFields().stream().anyMatch(field -> field.getType().isPresent())) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Excluded columns descriptor contains types");
+            }
+
+            // column names in DescriptorArgument are canonical.
+            // column names in TableArgument are not canonical, as they are taken from the corresponding RelationType.
+            // because of that, we match the excluded columns names case-insensitive
+            Set<String> excludedNames = excludedColumnsDescriptor.getFields().stream()
+                    .map(Descriptor.Field::getName)
+                    .map(name -> name.toLowerCase(ENGLISH))
+                    .collect(toImmutableSet());
+
+            List<RowType.Field> inputSchema = ((TableArgument) arguments.get(TABLE_ARGUMENT_NAME)).getRowType().getFields();
+            Set<String> inputNames = inputSchema.stream()
+                    .map(RowType.Field::getName)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .map(name -> name.toLowerCase(ENGLISH))
+                    .collect(toImmutableSet());
+
+            if (!inputNames.containsAll(excludedNames)) {
+                String missingColumns = Sets.difference(excludedNames, inputNames).stream()
+                        .collect(joining(", ", "[", "]"));
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, format("Excluded columns: %s not present in the table", missingColumns));
+            }
+
+            ImmutableList.Builder<Integer> requiredColumns = ImmutableList.builder();
+            ImmutableList.Builder<Descriptor.Field> returnedColumns = ImmutableList.builder();
+
+            for (int i = 0; i < inputSchema.size(); i++) {
+                Optional<String> name = inputSchema.get(i).getName();
+                if (name.isEmpty() || !excludedNames.contains(name.orElseThrow().toLowerCase(ENGLISH))) {
+                    requiredColumns.add(i);
+                    // per SQL standard, all columns produced by a table function must be named. It is required by the structure of Descriptor.
+                    returnedColumns.add(new Descriptor.Field(name.orElse(ANONYMOUS_COLUMN_NAME), Optional.of(inputSchema.get(i).getType())));
```

Comment:
@ebyhr I am aware that we assign names of the form `_col{n}` whenever we output an unnamed column. However, those names are inaccessible from within the query, and they organize the output with sequential distinct numbers (provided that the user did not explicitly use names like that).
If we used that same naming, it would clash:
```
trino:tiny> SELECT 0, 1, *
         -> FROM TABLE(exclude_columns(
         ->                    TABLE(SELECT * FROM (SELECT 3 a), (SELECT 4)),
         ->                    DESCRIPTOR(a)));
 _col0 | _col1 |  _col0
-------+-------+------------------
     0 |     1 |     4
(1 row)
```

In the first place, I think we should let table functions output anonymous columns. Currently they can't because so requires the SQL standard. However this is inconsistent with Trino and causes issues of this kind.